localization_manager/src/localization_manager_node.cpp

```diff
@@ -0,0 +1,172 @@
+
+        // Register runtime parameter update callback
+        add_on_set_parameters_callback(std::bind(&Node::parameter_update_callback, this, std_ph::_1));
+
+        // Initialize worker object
+        manager_.reset(new LocalizationManager(std::bind(&Node::publishPoseStamped, this, std_ph::_1),
+                                         std::bind(&Node::publishStatus, this, std_ph::_1),
+                                         std::bind(&Node::publishManagedInitialPose, this, std_ph::_1),
+                                         config_, 
+                                         get_node_logging_interface(),
+                                         std::make_unique<carma_ros2_utils::timers::ROSTimerFactory>(shared_from_this())
+                                         ));
+
+        // Setup subscribers
+        gnss_pose_sub_ = create_subscription<geometry_msgs::msg::PoseStamped>("gnss_pose", 5, 
+                                                                                std::bind(&LocalizationManager::gnssPoseCallback, manager_.get(), std_ph::_1));
+        initialpose_sub_ = create_subscription<geometry_msgs::msg::PoseWithCovarianceStamped>("initialpose", 1, 
+                                                                                            std::bind(&LocalizationManager::initialPoseCallback, manager_.get(), std_ph::_1));
+
+        // Setup synchronized message_filters subscribers
+        rclcpp::QoS qos(5);
+        ndt_pose_sub_.subscribe(this, "ndt_pose", qos.get_rmw_qos_profile());
+        ndt_score_sub_.subscribe(this,"ndt_stat", qos.get_rmw_qos_profile());
+
+        PoseStatsSynchronizer pose_stats_synchronizer(PoseStatsSyncPolicy(5), ndt_pose_sub_, ndt_score_sub_);
+        pose_stats_synchronizer.registerCallback(boost::bind(&Node::poseAndStatsCallback, this, _1, _2));
+        
+        // system_alert_topic_ protected member of CarmaLifecycleNode
+        system_alert_sub_ = create_subscription<carma_msgs::msg::SystemAlert>(system_alert_topic_, 1, 
+                                                                                std::bind(&LocalizationManager::systemAlertCallback, manager_.get(), std_ph::_1));
+        // Setup publishers
+        pose_pub_ = create_publisher<geometry_msgs::msg::PoseStamped>("selected_pose", 5);
+        state_pub_ = create_publisher<carma_localization_msgs::msg::LocalizationStatusReport>("localization_status", 5);
+        
+        // Create a publisher that will send all previously published messages to late-joining subscribers ONLY If the subscriber is transient_local too
+        rclcpp::PublisherOptions intra_proc_disabled; 
+        intra_proc_disabled.use_intra_process_comm = rclcpp::IntraProcessSetting::Disable; // Disable intra-process comms for this PublisherOptions object
+
+        auto pub_qos_transient_local = rclcpp::QoS(rclcpp::KeepAll()); // A publisher with this QoS will store all messages that it has sent on the topic
```

Comment:
For managed initial_pose a KeepLast(1) makes more sense than KeepAll. We wouldn't want to send multiple initial poses in a row on new connection. We only send the most recent.